src/course-unit/data/thunk.js

```diff
@@ -239,3 +246,37 @@ export function fetchCourseVerticalChildrenData(itemId) {
     }
   };
 }
+
+export function copyToClipboard(usageKey) {
+  const POLL_INTERVAL_MS = 1000; // Timeout duration for polling in milliseconds
+
+  return async (dispatch) => {
+    dispatch(updateClipboardData(null));
+    dispatch(showProcessingNotification(NOTIFICATION_MESSAGES.copying));
+    dispatch(updateSavingStatus({ status: RequestStatus.IN_PROGRESS }));
+    dispatch(updateQueryPendingStatus(true));
```

Comment:
[important]: Currently, the notification about Internet problems does not work as expected (an error message and a message about the lack of Internet connection appear). Note how this works for the functionality of adding a new unit.

https://github.com/raccoongang/frontend-app-course-authoring/assets/93188219/18a94a5e-1ee2-4372-a479-135dc0378dc7